### PR TITLE
Github user can add an exist site from a list

### PR DESCRIPTION
**Pull request by singpolyma**
Fetch the list of repos from Github, display the list, add site by
hitting a button.

Part of #38

### CODE REVIEW

File: app/git_providers/github.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+				end
+
+				results
+			end
+
+		protected
+
+			def repositories_page(uri)
+				resp = request("GET", uri, per_page: 100)
+				raise resp unless resp.is_a?(Net::HTTPOK)
```
Comment (ericroberts) [inline app/git_providers/github.rb]: When testing this I got a response of class `Net::HTTPUnauthorized` but it did not raise the exception as expected. It is a TypeError with the message "exception class/object expected". We could get around this with a custom exception class that takes in the response.

**Comment by ericroberts** [comment]
Whenever I try this I am getting Net::HTTPUnauthorized as a response. According to omniauth my token doesn't expire, so I am not sure why this is happening.

**Comment by ericroberts** [comment]
Have you considered using this gem? https://github.com/piotrmurach/github. Our stuff is pretty simple here so far, but maybe something to think about if we think our github use will become more complex. We could still use your wrapping class to call into the gem code.

**Comment by ericroberts** [comment]
We should update the login code to update the omniauth info on login, because otherwise we could have users with expired tokens and this will crash for them.

File: app/controllers/site.rb [hunk 2]
```diff
@@ -13,7 +14,14 @@ class Site < Base
 					slim :"site/index", locals: { site: site }
 				else
 					authorize Tint::Site, :index?
-					slim :index, locals: { sites: policy_scope(Tint::Site) }
+
+					git_providers = DB[:identities].where(user_id: pundit_user[:user_id]).map do |identity|
+						if identity[:provider] == "github"
```
Comment (ericroberts) [inline app/controllers/site.rb]: Use `select` instead and then we can remove the compact call?

**Comment by singpolyma** [comment]
@ericroberts ready for review again

**Pull request merged**